src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3ResponseService.java

```diff
@@ -63,4 +66,28 @@ public void saveEt3ResponseDocument(CaseData caseData, DocumentInfo documentInfo
 
         caseData.getDocumentCollection().add(documentTypeItem);
     }
-}
\ No newline at end of file
+
+    public void saveRelatedDocumentsToDocumentCollection(CaseData caseData) {
+        if (CollectionUtils.isEmpty(caseData.getDocumentCollection())) {
+            caseData.setDocumentCollection(new ArrayList<>());
+        }
+
+        List<DocumentTypeItem> documents = caseData.getDocumentCollection();
+
+        documents.addAll(
+                Optional.ofNullable(caseData.getEt3ResponseContestClaimDocument())
+                .orElse(List.of())
+                .stream()
+                .filter(o -> documents.stream().noneMatch(x -> x.getId().equals(o.getId())))
```

Comment:
Stream within a Stream? is there anyway to be more clear whats going on here?

Reply:
Good point, it might be cleaner (and more efficient) to use a map of all IDs and filter by documents whose id isn't already in the map.